File: src/main/java/ru/ezhov/rocket/action/configuration/ui/NewRocketActionSettings.java
package ru.ezhov.rocket.action.configuration.ui;

import ru.ezhov.rocket.action.api.RocketActionSettings;

import java.util.List;
import java.util.Map;
import java.util.UUID;

public class NewRocketActionSettings implements RocketActionSettings {
    private String id;
    private String type;
    private Map<String, String> settings;
    private List<RocketActionSettings> actions;

    public NewRocketActionSettings(
            String type,
            Map<String, String> settings,
            List<RocketActionSettings> actions
    ) {
        this.type = type;
        this.settings = settings;
        this.actions = actions;
        this.id = UUID.randomUUID().toString();
    }

    @Override
    public String id() {
        return null;
    }

    @Override
    public String type() {
        return type;
    }

    @Override
    public Map<String, String> settings() {
        return settings;
    }

    @Override
    public List<RocketActionSettings> actions() {
        return actions;
    }
}

File: src/main/kotlin/ru/ezhov/rocket/action/configuration/ui/NewRocketActionSettings.kt
package ru.ezhov.rocket.action.configuration.ui

import ru.ezhov.rocket.action.api.RocketActionConfiguration
import ru.ezhov.rocket.action.api.RocketActionConfigurationPropertyKey
import ru.ezhov.rocket.action.api.RocketActionSettings
import ru.ezhov.rocket.action.api.RocketActionType
import java.util.UUID

class NewRocketActionSettings(
    val configuration: RocketActionConfiguration,
    private val type: RocketActionType,
    private val settings: Map<RocketActionConfigurationPropertyKey, String>,
    private val actions: List<RocketActionSettings> = emptyList()
) : RocketActionSettings {
    private val id: String = UUID.randomUUID().toString()

    override fun id(): String = id

    override fun type(): RocketActionType = type

    override fun settings(): Map<RocketActionConfigurationPropertyKey, String> = settings

    override fun actions(): List<RocketActionSettings> = actions
}